ceremonies/src/lib.rs

```diff
@@ -1946,6 +2006,34 @@ impl<T: Config> Pallet<T> {
 		));
 		Ok(())
 	}
+
+	fn has_reputation(participant: &T::AccountId, cid: &CommunityIdentifier) -> bool {
+		let reputation_lifetime = Self::reputation_lifetime();
+		let cindex = <encointer_scheduler::Pallet<T>>::current_ceremony_index();
+		for i in 0..=reputation_lifetime {
+			if Self::participant_reputation(&(*cid, cindex.saturating_sub(i)), participant)
+				.is_verified()
+			{
+				return true
+			}
+		}
+		false
+	}
+
+	fn is_endorsed(
+		participant: &T::AccountId,
+		cc: &CommunityCeremony,
+	) -> Option<CeremonyIndexType> {
+		let reputation_lifetime = Self::reputation_lifetime();
+		for i in (0..=reputation_lifetime).rev() {
```

Comment:
Big props for the `rev()` for optimization! A comment for the people outside the occult circle to understand would be nice though.

Reply:
I think you are giving the props for the wrong thing😅 I use rev in order that the oldest endorsement will be detected, and i didnt have any optimizations in mind. what optimization is it you are talking about?

Reply:
Haha, well in that case I think that you actually counter optimized. :P Because the closes you are to the current ceremony index, the more likely it is that you have the endorsement for it. I misread the code, so I thought that it checks the newest index first.

Hmm, doesn't the code disallow multiple endorsements?

Reply:
Hmm, as long as we have a small reputation lifetime this is fine, but the dev-network might loop too long if we do it like this. Reversing the order does not resolve that issue however if the endorsee is not registered at all.

In general, I believe we should consider adding a safety parameter that prevents big loops if the reputation lifetime is large. Or, on the other hand, this is why we might want to explicitly pass the reputations after all. But in this specific case, we do not know which ceremony indexes we need to look at. Hmmm, parametrized for loops in the runtime are scary.

For now I suggest we should reverse the order to optimize. Re-thinking for loops we can do at another time.